packages/perseus-editor/src/components/graph-locked-figures/locked-function-settings.tsx

```diff
@@ -0,0 +1,246 @@
+    //     entering in a negative value - used when specifying value of input field
+    const [domainEntries, setDomainEntries] = useState([
+        domain && domain[0] !== -Infinity ? domain[0].toString() : "",
+        domain && domain[1] !== Infinity ? domain[1].toString() : "",
+    ]);
+
+    useEffect(() => {
+        setDomainEntries([
+            domain && domain[0] !== -Infinity ? domain[0].toString() : "",
+            domain && domain[1] !== Infinity ? domain[1].toString() : "",
+        ]);
+    }, [domain]);
+
+    // Generic function for handling property changes (except for 'domain')
+    function handlePropChange(property: string, newValue: string) {
+        const updatedProps: Partial<LockedFunctionType> = {};
+        updatedProps[property] = newValue;
+        onChangeProps(updatedProps);
+    }
+
+    /*
+     Reason for having a separate 'propChange' function for 'domain':
+        Domain entries are optional. Their default value is +/- Infinity.
+        Since input fields that are empty evaluate to zero, there needs to be
+            dedicated code to convert empty to Infinity.
+     */
+    function handleDomainChange(limitIndex: number, newValueString: string) {
+        const newDomainEntries = [...domainEntries];
+        newDomainEntries[limitIndex] = newValueString;
+        setDomainEntries(newDomainEntries);
+        const newDomain: Interval | undefined = [
+            domainLimits[0],
+            domainLimits[1],
+        ];
+
+        let newValue = parseFloat(newValueString);
+        if (newValueString === "" && limitIndex === 0) {
+            newValue = -Infinity;
+        } else if (newValueString === "" && limitIndex === 1) {
+            newValue = Infinity;
+        }
+        newDomain[limitIndex] = newValue;
+        onChangeProps({domain: newDomain});
+    }
+
+    return (
+        <UniqueIDProvider
+            mockOnFirstRender={true}
+            scope="locked-function-settings"
+        >
+            {(ids) => (
+                <PerseusEditorAccordion
+                    expanded={props.expanded}
+                    onToggle={props.onToggle}
+                    header={
+                        <View style={styles.row}>
+                            <LabelLarge style={styles.accordionHeader}>
+                                {lineLabel}
+                            </LabelLarge>
+                            <Strut size={spacing.xSmall_8} />
+                            <LineSwatch
+                                color={lineColor}
+                                lineStyle={strokeStyle}
+                            />
+                        </View>
+                    }
+                >
+                    <View style={[styles.row, styles.spaceUnder]}>
+                        {/* Line color settings */}
+                        <ColorSelect
+                            selectedValue={lineColor}
+                            onChange={handlePropChange.bind(null, "color")}
```

Comment:
Nit: I think all the other `onChange` props across the interactive graph editor use the arrow function notation instead. I'd probably switch over to that for consistency's sake.

```
onChange={(value) => handlePropChange("color", value)}
```

I also find that this is a lot more explicit about what it's doing compared to having to pass `null` into the `bind()`.